ui/desktop/electron-app/src/helpers/app-updater.js

```diff
@@ -61,9 +61,11 @@ const findUpdateArchive = (version) => {
   const url = `${releasesUrl}${version}/boundary-desktop_${version}_darwin_${architecture}.zip`;
   return new Promise((resolve, reject) => {
     https.get(url, (response) => {
-      if (response.statusCode === 403)
```

Comment:
We should be returning an error on any non `200` response. This likely made it harder for us to debug as we couldn't see this error out as it was a `404` response.